src/tui/mainview.zig

```diff
@@ -550,6 +550,40 @@ const cmds = struct {
     }
     pub const add_diagnostic_meta = .{ .arguments = &.{ .string, .string, .string, .string, .integer, .integer, .integer, .integer, .integer } };
 
+    pub fn rename_symbol_item(self: *Self, ctx: Ctx) Result {
+        // because the incoming message is an array of Renames, we manuallly
+        // parse instead of using ctx.args.match() which doesn't seem to return
+        // the parsed length needed to correctly advance iter.
+        var iter = ctx.args.buf;
+        var len = try cbor.decodeArrayHeader(&iter);
+        var mroot: ?@import("Buffer").Root = null;
+        while (len != 0) {
+            var file_uri: []const u8 = undefined;
+            var sel: ed.Selection = .{};
+            var new_text: []const u8 = undefined;
+            len -= 1;
+            std.debug.assert(try cbor.decodeArrayHeader(&iter) == 6);
+            if (!try cbor.matchString(&iter, &file_uri)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.begin.row)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.begin.col)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.end.row)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.end.col)) return error.MissingArgument;
+            if (!try cbor.matchString(&iter, &new_text)) return error.MissingArgument;
+            file_uri = project_manager.normalize_file_path(file_uri);
+            if (self.get_active_editor()) |editor| {
+                // TODO match file_uri correctly. endsWith() isn't correct because 'path' is a
+                // short, relative path while 'file_uri' is an absolute path starting with 'file://'
+                const match = if (editor.file_path) |path| std.mem.endsWith(u8, file_uri, path) else false;
+                if (match) {
+                    try editor.rename_symbol_item(sel, new_text, &mroot, len == 0);
+                } else {
+                    log.logger("LSP").print("TODO perform renames in other files\n", .{});
+                }
+            }
+        }
+    }
+    pub const rename_symbol_item_meta = .{ .arguments = &.{ .string, .integer, .integer, .integer, .integer, .string } };
+
```

Comment:
This is outdated.  The rename message is now an array of these.  But I'm not sure how to express that.  Would this be correct?
```zig
.arguments = &.{ .array }
```

Reply:
Yes, array sounds fine. You can add `array` to the `ArgumentType` enum.

Reply:
Thanks. I've updated this just now and renamed the old one to `rename_symbol_item_elem_meta` since just `.array` loses some info.  Let me know if I should remove it.